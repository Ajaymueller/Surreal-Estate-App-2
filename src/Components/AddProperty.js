import React, { useState } from "react";
import Alert from "./Alert";
import PropertyForm from "./PropertyForm";
import { saveProperty } from "../requests";

import "../styles/AddProperty.css";

const initialState = {
  fields: {
    title: "",
    city: "Manchester",
    type: "",
    bedrooms: 0,
    bathrooms: 0,
    price: 0,
    email: "",
  },
  alert: {
    message: "",
    isSuccess: false,
  },
};

const AddProperty = () => {
  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);

  const handleAddProperty = (event) => {
    event.preventDefault();
    setAlert({ message: "", isSuccess: false });

    return saveProperty(fields, setAlert);
  };

  const handleFieldChange = (event) => {
    const changedField = event.target.name;
    const newValue = event.target.value;

    setFields({ ...fields, [changedField]: newValue });
  };

  return (
    <div className="add-property" data-testid="addProperty">
       <PropertyForm 
        handleAddProperty={handleAddProperty}
        handleFieldChange={handleFieldChange}
        title={fields.title}
        city={fields.city}
        type={fields.type}
        bedrooms={fields.bedrooms}
        bathrooms={fields.bathrooms}
        price={fields.price}
        email={fields.email}
        />
      <Alert message={alert.message} success={alert.isSuccess} />
    </div>
  );
};

export default AddProperty;