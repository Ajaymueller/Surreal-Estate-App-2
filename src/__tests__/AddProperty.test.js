import React from 'react';
import { render } from '@testing-library/react';
import AddProperty from '../Components/AddProperty';
import { MemoryRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';

describe("AddProperty", () => {
    it("component renders correctly", () => {
        const { asFragment } = render (
        <MemoryRouter><AddProperty /></MemoryRouter> )
        expect(asFragment).toMatchSnapshot();
    });
    it("returns a div with PropertyForm and Alert component", () => {
        const { getByTestId } = render (
        <MemoryRouter><AddProperty /></MemoryRouter> )
        const propertyForm = getByTestId("addProperty");
        expect(propertyForm).toBeInTheDocument();
        expect(propertyForm).toHaveClass("add-property");
    });
});