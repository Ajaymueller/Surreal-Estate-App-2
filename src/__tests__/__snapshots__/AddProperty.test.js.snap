exports[`AddProperty component renders correctly 1`] = `[Function]`;
